cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/transform/ScriptTransform.java

```diff
@@ -258,10 +269,21 @@ private Object decodeUnion(Object object, List<Schema> schemas) {
     throw new RuntimeException("Unable decode union with schema " + schemas);
   }
 
-  private void init() {
+  private void init(LookupProvider lookup) {
     ScriptEngineManager manager = new ScriptEngineManager();
     engine = manager.getEngineByName("JavaScript");
-    engine.put(CONTEXT_NAME, new ScriptContext(logger, metrics));
+    try {
+      engine.eval(ScriptConstants.HELPER_DEFINITION);
+    } catch (ScriptException e) {
+      // shouldn't happen
+      throw new IllegalStateException("Couldn't define helper functions", e);
+    }
+
+    JavaTypeConverters js = ((Invocable) engine).getInterface(
+      engine.get(ScriptConstants.HELPER_NAME), JavaTypeConverters.class);
+
+    engine.put(CONTEXT_NAME, new ScriptContext(
+      logger, metrics, lookup, GSON.fromJson(config.lookup, LookupConfig.class), js));
```

Comment:
i think we should have parsing the JSON in a try-catch block similar to what we do for parsing ``config.schema`` JSON below and throw appropriate exception..